src/deepsparse/transformers/utils/decoder_kv_cache.py

```diff
@@ -103,8 +99,12 @@ def update(
         :param input_ids_len: The number of input ids in the current
             input batch: (batch_size, length).
             Corresponds to `input_ids.shape[1]`
+        :param increment_total_num_processed_tokens: If set to True,
+            the total number of processed tokens will be incremented
+            by the input_ids_len.
         """
-        self.total_num_processed_tokens += input_ids_len
+        if increment_total_num_processed_tokens:
+            self.total_num_processed_tokens += input_ids_len
```

Comment:
Don't we always want to keep self.total_num_processed_tokens up to date? What is the use case for making this conditional?

Reply:
This boolean is there to allow self.update method (that is by default responsible for "shrinking" the new kv cache after forward pass) to be also reused when we decrease the capacity of the kv cache.
In the first scenario, we want to increment total_num_processed_tokens.
In the second scenario, we only want to manipulate the shape of the KV cache so it fits the assumptions of the next engine that will digest it. It would be incorrect to increment total_num_processed_tokens in this scenario.